target/linux/generic/pending-5.15/833-1-dt-bindings-leds-sort-LED-color-definitions-by-alpha.patch

```diff
@@ -0,0 +1,54 @@
+From: Shiji Yang <yangshiji66@outlook.com>
+Date: Tue, 31 Oct 2023 16:46:40 +0800
+Subject: [PATCH 1/3] dt-bindings: leds: sort LED color definitions by alphabet
+
+Sorting LEDs by alphabet can help developers quickly find the colors
+they want. Infrared and multicolor LEDs are special so they should
+be left at the end of the list.
+
+Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
+---
+ include/dt-bindings/leds/common.h | 31 ++++++++++++++++---------------
+ 1 file changed, 16 insertions(+), 15 deletions(-)
+
+--- a/include/dt-bindings/leds/common.h
++++ b/include/dt-bindings/leds/common.h
+@@ -22,22 +22,23 @@
+ #define LEDS_BOOST_FIXED	2
+ 
+ /* Standard LED colors */
+-#define LED_COLOR_ID_WHITE	0
+-#define LED_COLOR_ID_RED	1
+-#define LED_COLOR_ID_GREEN	2
+-#define LED_COLOR_ID_BLUE	3
+-#define LED_COLOR_ID_AMBER	4
+-#define LED_COLOR_ID_VIOLET	5
+-#define LED_COLOR_ID_YELLOW	6
+-#define LED_COLOR_ID_IR		7
+-#define LED_COLOR_ID_MULTI	8	/* For multicolor LEDs */
+-#define LED_COLOR_ID_RGB	9	/* For multicolor LEDs that can do arbitrary color,
++#define LED_COLOR_ID_AMBER	0
```

Comment:
This will change the DTB ABI. For OpenWrt it should be fine as we build our DTB together with the kernel, but if someone wants to update the dtb independent from the kernel this could cause problems. I think the upstream kernel is not so strict about DTB ABI breakage any more, but I think we should not do them if they are not adding a clear benefit.

Reply:
If you get this accepted upstream I am fine with this change, but it could get challenging to get this into mainline Linux kernel.

Reply:
same... don't think it will be accepted and IMHO it would make more harm than good... we use define for the exact reason of not caring about order of actual value

Reply:
Haha, it has been rejected. I will remove it in the next push.